extendedcopy_test.go

```diff
@@ -675,6 +675,120 @@ func TestExtendedCopyGraph_FilterAnnotationWithRegex(t *testing.T) {
 	verifyCopy(dst, copiedIndice, uncopiedIndice)
 }
 
+func TestExtendedCopyGraph_FilterAnnotationWithNoKey(t *testing.T) {
+	// generate test content
+	var blobs [][]byte
+	var descs []ocispec.Descriptor
+	appendBlob := func(mediaType string, blob []byte, key string, value string) {
+		blobs = append(blobs, blob)
+		descs = append(descs, ocispec.Descriptor{
+			MediaType:   mediaType,
+			Digest:      digest.FromBytes(blob),
+			Size:        int64(len(blob)),
+			Annotations: map[string]string{key: value},
+		})
+	}
+	generateArtifactManifest := func(subject ocispec.Descriptor, key string, value string) {
+		var manifest artifactspec.Manifest
+		artifactSubject := descriptor.OCIToArtifact(subject)
+		manifest.Subject = &artifactSubject
+		manifest.Annotations = map[string]string{key: value}
+		manifestJSON, err := json.Marshal(manifest)
+		if err != nil {
+			t.Fatal(err)
+		}
+		appendBlob(artifactspec.MediaTypeArtifactManifest, manifestJSON, key, value)
+	}
+	appendBlob(ocispec.MediaTypeImageLayer, []byte("foo"), "bar", "blackpink") // descs[0]
+	generateArtifactManifest(descs[0], "bar", "bluebrown")                     // descs[1]
+	generateArtifactManifest(descs[0], "bar", "blackred")                      // descs[2]
+	generateArtifactManifest(descs[0], "bar", "blackviolet")                   // descs[3]
+	generateArtifactManifest(descs[0], "bar", "greengrey")                     // descs[4]
+	generateArtifactManifest(descs[0], "bar", "brownblack")                    // descs[5]
+	ctx := context.Background()
+	src := memory.New()
+	for i := range blobs {
+		err := src.Push(ctx, descs[i], bytes.NewReader(blobs[i]))
+		if err != nil {
+			t.Fatalf("failed to push test content to src: %d: %v", i, err)
+		}
+	}
+	opts := oras.ExtendedCopyGraphOptions{}
+	exp := "black."
+	regex := regexp.MustCompile(exp)
+	opts.FilterAnnotation("", regex)
+	if opts.FindPredecessors != nil {
+		t.Fatal("FindPredecessors not nil!")
+	}
+}
+
+func TestExtendedCopyGraph_FilterAnnotationWithNoRegex(t *testing.T) {
```

Comment:
Maybe we can merge `TestExtendedCopyGraph_FilterAnnotationWithNoKey` and `TestExtendedCopyGraph_FilterAnnotationWithNoRegex` into `TestExtendedCopyGraph_FilterAnnotation`?